tuf/api/_payload.py

```diff
@@ -731,21 +755,16 @@ def _get_length_and_hashes(
         hashes = {}
 
         if hash_algorithms is None:
-            hash_algorithms = [sslib_hash.DEFAULT_HASH_ALGORITHM]
+            hash_algorithms = [_DEFAULT_HASH_ALGORITHM]
 
         for algorithm in hash_algorithms:
             try:
                 if isinstance(data, bytes):
-                    digest_object = sslib_hash.digest(algorithm)
+                    digest_object = _hash(algorithm)
                     digest_object.update(data)
                 else:
-                    digest_object = sslib_hash.digest_fileobject(
-                        data, algorithm
-                    )
-            except (
-                sslib_exceptions.UnsupportedAlgorithmError,
-                sslib_exceptions.FormatError,
-            ) as e:
+                    digest_object = _file_hash(data, algorithm)
+            except (ValueError, TypeError) as e:
```

Comment:
the different functionality is a little annoying here (you need to call update on one digest_object but not the other)... 
* If the `_hash` and `_file_hash` functions directly returned the hexdigest this code would be simpler and the functions would be properly annotated
* on the other hand then _hash() would need a data argument (meaning you would have to split the lambda needed by hashlib.file_digest() into another function)

I don't really know what is the simplest... up to you

Reply:
Does [535a189](https://github.com/theupdateframework/python-tuf/pull/2815/commits/535a18918bff8896642ac13602aa3c53e3bdb14b) look better? I don't have very strong feelings about either way.

Reply:
No passionate argument here either.  I think I like this new version better.